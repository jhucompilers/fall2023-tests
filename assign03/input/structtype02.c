struct Point {
  int x, y;
};

struct Player {
  struct Point location;
  char symbol;
  short health;
};

int main(void) {
  struct Player p;

  p.location.x = 3;
  p.location.y = 4;
  p.symbol = '@';
  p.health = 100;

  return 0;
}
