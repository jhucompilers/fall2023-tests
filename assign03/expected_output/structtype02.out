0|struct Point|type|struct Point {}
1|x|variable|int
1|y|variable|int
0|struct Player|type|struct Player {}
1|location|variable|struct Point {int, int}
1|symbol|variable|char
1|health|variable|short
0|main|function|function () returning int
2|p|variable|struct Player {struct Point {int, int}, char, short}
